### PR TITLE
Improve scene_exceptions

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by fernandog** [comment]
@medariox 

_________________________ FLAKE8-check(ignoring D100) __________________________
/home/travis/build/pymedusa/SickRage/sickbeard/scene_exceptions.py:31:1: I100 Import statements are in the wrong order.from six should be before from sickbeard.indexers.indexer_config

**Comment by fernandog** [comment]
@medariox still in progress?

**Comment by medariox** [comment]
@fernandog 
Yeah, still working on it. I fixed most issues, but need some more time to refactor the code.

**Comment by fernandog** [comment]
@medariox milestone 0.1.5 then?

**Comment by medariox** [comment]
This PR didn't make it as far as I'd have wished. Don't really have much time to clean up the rest of the code and since it should solve [this bug](https://github.com/pymedusa/SickRage/issues/901) as well, I'll leave it for now.

@fernandog Can you please rebase?

**Comment by fernandog** [comment]
no one else is going to test this. imo it should be merged to develop so everybody test it

**Comment by fernandog** [comment]
@medariox why not always erase scene_exceptions before adding new ones (if succeed to fetch from github)? What do you think?

File: sickbeard/scene_exceptions.py [hunk 1]
```diff
@@ -1,341 +1,350 @@
...
 
-def retrieve_exceptions():  # pylint:disable=too-many-locals, too-many-branches
-    """
-    Looks up the exceptions on github, parses them into a dict, and inserts them into the
-    scene_exceptions table in cache.db. Also clears the scene name cache.
-    """
+def update_scene_exceptions(indexer_id, scene_exceptions, season=-1):
+    """Given a indexer_id, and a list of all show scene exceptions, update the db."""
+    cache_db_con = db.DBConnection('cache.db')
+    cache_db_con.action(b'DELETE FROM scene_exceptions WHERE indexer_id=? and season=?', [indexer_id, season])
```
Comment (fernandog) [inline sickbeard/scene_exceptions.py]: @medariox seems we already does it here.

so don't understand why we had duplicates

**Comment by medariox** [comment]
@fernandog 
I'm not sure about this. Is this safe?

**Comment by fernandog** [comment]
yes, it is

**Comment by fernandog** [comment]
Approved

**Pull request merged**